Wordle.Api/Wordle.Api/Controllers/TokenController.cs

```diff
@@ -0,0 +1,135 @@
+using System.Security.Claims;
+using System.Text;
+using Wordle.Api.Data;
+using Wordle.Api.Identity;
+using Wordle.Api.Models;
+
+namespace Wordle.Api.Controllers;
+[Route("Token")]
+[ApiController]
+public class TokenController : Controller
+{
+    public AppDbContext _context;
+    public UserManager<AppUser> _userManager;
+    public JwtConfiguration _jwtConfiguration;
+    public TokenController(AppDbContext context, UserManager<AppUser> userManager, JwtConfiguration jwtConfiguration)
+    {
+        _context = context;
+        _userManager = userManager;
+        _jwtConfiguration = jwtConfiguration;
+    }
+    [HttpPost("GetToken")]
+    public async Task<IActionResult> GetToken([FromBody] UserCredentials userCredentials)
+    {
+        if (string.IsNullOrEmpty(userCredentials.Username))
+        {
+            return BadRequest("Username is required");
+        }
+        if (string.IsNullOrEmpty(userCredentials.Password))
+        {
+            return BadRequest("Password is required");
+        }
+
+        var user = _context.Users.FirstOrDefault(u => u.UserName == userCredentials.Username);
+
+        if (user is null) { return Unauthorized("The user account was not found"); }
+
+        bool results = await _userManager.CheckPasswordAsync(user, userCredentials.Password);
+        if (results)
+        {
+            var securityKey = new SymmetricSecurityKey(Encoding.UTF8.GetBytes(_jwtConfiguration.Secret));
+            var credentials = new SigningCredentials(securityKey, SecurityAlgorithms.HmacSha512);
+
+            var claims = new List<Claim>
+            {
+                new Claim(JwtRegisteredClaimNames.Sub, user.UserName!),
+                new Claim(JwtRegisteredClaimNames.Jti, Guid.NewGuid().ToString()),
+                new Claim(Claims.UserId, user.Id.ToString()),
+                new Claim(Claims.Random, (new Random()).NextDouble().ToString()),
+                new Claim(Claims.UserName, user.UserName!.ToString().Substring(0,user.UserName.ToString().IndexOf("@"))),
+            };
+            var roles = await _userManager.GetRolesAsync(user);
+            foreach (var role in roles)
+            {
+                claims.Add(new Claim(ClaimTypes.Role, role));
+            }
+
+            var token = new JwtSecurityToken(
+                issuer: _jwtConfiguration.Issuer,
+                audience: _jwtConfiguration.Audience,
+                claims: claims,
+                expires: DateTime.Now.AddMinutes(_jwtConfiguration.ExpirationInMinutes),
+                signingCredentials: credentials
+            );
+            var jwtToken = new JwtSecurityTokenHandler().WriteToken(token);
+            return Ok(new { token = jwtToken });
+        }
+        return Unauthorized("The username or password is incorrect");
+    }
+
+    [HttpPost("CreateUser")]
+    public async Task<IActionResult> CreateUser([FromBody]CreateUser createUser)
+    {
+        if (string.IsNullOrEmpty(createUser.Username))
```

Comment:
I suggest you check and make sure that the username contains an @, it will make problems otherwise